src/vm_pages.rs

```diff
@@ -595,6 +665,8 @@ pub struct VmPages<T: GuestStagePageTable, S = VmStateFinalized> {
     root: PlatformPageTable<T>,
     measurement: Mutex<Sha256Measure>,
     pte_pages: PtePagePool,
+    confidential_page_range: Mutex<Option<PageRangeInfo>>,
```

Comment:
Limiting a TVM to a single contiguous confidential page range seems overly restrictive.

Reply:
I am open to suggestions, but think that we do need to have a single contiguous range to be able to efficiently search the GPA-map when DefineSharedPagesRange is called. Basically, we want to quickly rule out the possibility that the shared page range doesn't overlap with any confidential address space. To give an extreme example, suppose that the host defines a confidential page at the bottom and top of the GPA, and then tries to define a shared address space somewhere in the middle. Unless we maintain some sort of sparse bit vector to allow us to identify where confidential pages are mapped, it's computationally expensive to perform a search of the GPA space.